v2/src/datadog.ts

```diff
@@ -23,7 +23,7 @@ import {
   applyEnvVariables,
   validateProps,
   TagKeys,
-  IDatadogProps,
+  DatadogPropsV2,
```

Comment:
we need to include ISecret (specific to CDK v2) in our props interface. Using type intersection (`&`) doesn't work with JSII (the bug introduced that this PR is fixing). Therefore we need to create a v2 specific interface that includes ISecret.